src/Php/PhpSettingsHandler.php

```diff
@@ -0,0 +1,418 @@
+
+/**
+ * @author John Stevenson <john-stevenson@blueyonder.co.uk>
+ * @author Théo Fidry <theo.fidry@gmail.com>
+ */
+class PhpSettingsHandler
+{
+    // MODIFIED
+    const SUFFIX_ALLOW = '_ALLOW_INITIAL_SETTINGS';
+    // --MODIFIED
+    const SUFFIX_INIS = '_ORIGINAL_INIS';
+    const RESTART_ID = 'internal';
+
+    private static $name;
+    private static $skipped;
+
+    private $envPrefix;
+    private $cli;
+    private $colorOption;
+    // MODIFIED: varaible name
+    private $envAllowIniSettings;
+    // --MODIFIED
+    private $envOriginalInis;
+    private $restart;
+    private $statusWriter;
+    private $tmpIni;
+
+    /**
+     * Constructor
+     *
+     * The $envPrefix is used to create distinct environment variables. It is
+     * uppercased and prepended to the default base values. For example 'myapp'
+     * would result in MYAPP_ALLOW_XDEBUG and MYAPP_ORIGINAL_INIS.
+     *
+     * @param string $envPrefix Value used in environment variables
+     * @param string $colorOption Command-line long option to force color output
+     * @param LoggerInterface $logger Activates status message output to a PSR-3 logger
+     *
+     * @throws RuntimeException If a parameter is invalid
+     */
+    final public function __construct($envPrefix, $colorOption = '', LoggerInterface $logger = null)
+    {
+        // MODIFIED: function made final + logger
+        if (!is_string($envPrefix) || empty($envPrefix) || !is_string($colorOption)) {
+            throw new RuntimeException('Invalid constructor parameter');
+        }
+
+        self::$name = strtoupper($envPrefix);
+
+        $this->envPrefix = $envPrefix;
+        $this->envAllowIniSettings = self::$name.self::SUFFIX_ALLOW;
+        $this->envOriginalInis = self::$name.self::SUFFIX_INIS;
+
+        $this->colorOption = $colorOption;
+        $this->cli = PHP_SAPI === 'cli';
+
+        // MODIFIED
+        // renamed $this->loaded to $this->restart
+        $this->restart = $this->shouldRestart();
+
+        if (null !== $logger) {
+            $this->setLogger($logger);
+        }
+        // -- MODIFIED
+    }
+
+
+    /**
+     * @return bool
+     */
+    protected function shouldRestart()
+    {
+        return extension_loaded('xdebug') || in_array(ini_get('phar.readonly'), ['1', 'On'], true);
+    }
+
+    /**
+     * @return array All current ini settings
+     */
+    protected function getIniSettings()
+    {
+        $loaded = ini_get_all(null, false);
+        $loaded['phar.readonly'] = '0';
+
+        return $loaded;
+    }
+
+    /**
+     * Returns true if the tmp ini file was written
+     *
+     * The filename is passed as the -c option when the process restarts.
+     *
+     * @param array $iniFiles All ini files used in the current process
+     *
+     * @return bool
+     */
+    private function writeTmpIni(array $iniFiles)
+    {
+        // MODIFIED
+        if (!$this->tmpIni = tempnam(sys_get_temp_dir(), $this->envPrefix.'_')) {
```

Comment:
Oh I see. Well my train of though is that I wanted to be able to identify them to double check they were removed properly or something. Since it's for security reasons though we can disregard that change completely